File: smartcontent-frontend/src/services/aiService.js
import api from './api';

export const aiService = {
  generateSummary: async (content) => {
    const response = await api.post('/ai/summarize', { content });
    return response;
  },

  generateTags: async (title, content) => {
    const response = await api.post('/ai/generate-tags', { title, content });
    return response;
  },

  calculateSeoScore: async (title, content) => {
    const response = await api.post('/ai/seo-score', { title, content });
    return response;
  },
};

export default aiService;

File: smartcontent-backend/src/main/java/com/smartcontent/service/AiService.java
package com.smartcontent.service;

import com.smartcontent.dto.AiMetadataDto;

public interface AiService {

    AiMetadataDto.SummarizationResponse generateSummary(String content);

    AiMetadataDto.TagGenerationResponse generateTags(String title, String content);

    AiMetadataDto.SeoScoreResponse calculateSeoScore(String title, String content);
}
